packages/server/src/queries/complex/swap.ts

```diff
@@ -0,0 +1,200 @@
+import { CoinPretty, Int, RatePretty } from "@keplr-wallet/unit";
+import { SplitTokenInQuote, SplitTokenOutQuote } from "@osmosis-labs/pools";
+import { AssetList } from "@osmosis-labs/types";
+import { z } from "zod";
+
+import { captureIfError } from "../../utils";
+import { getSidecarRouter } from "../sidecar/router";
+import { getAsset, getCosmwasmPoolTypeFromCodeId, Pool } from ".";
+
+export const getRouteTokenOutGivenInParams = z.object({
+  tokenInDenom: z.string(),
+  tokenInAmount: z.string(),
+  tokenOutDenom: z.string(),
+  forcePoolId: z.string().optional(),
+});
+
+type GetRouteTokenOutGivenInParams = z.infer<
+  typeof getRouteTokenOutGivenInParams
+>;
+
+export async function getRouteTokenOutGivenIn({
```

Comment:
Are these driveby changes? May want to work coordinate w @mattupham, may be conflicts here

Reply:
@mattupham This [comment](https://github.com/osmosis-labs/osmosis-frontend/pull/3885#discussion_r1805490532) should resolve this issue.